### ISSUE TITLE
Missing one image with point_triangulator & error with 2 images. 

### ISSUE THREAD

**Issue by mjmjeong** [colmap/colmap#1291]
**1. Missing one image's information when using colmap point_triangulator** (Sparse reconstruction with known camera pose)

To check the working of point_triangulator function, I conducted the following steps.    

- Sparse reconstruction(using colmap mapper)
  This step makes camera poses and points3D of all images. 
  
- Extract camera pose from the sparse reconstruction.
 ```bash
colmap model_converter \
    --input_path $PROJECT_PATH/sparse/ \
    --output_path $PROJECT_PATH/inputs/ \
    --output_type TXT
```
- Modifying `$PROJECT_PATH/inputs` directory
   I make `points3D.txt` file blank, and remove the 3d points information in `images.txt` as following image. I already checked the image-id is same with database.db.

![image](https://user-images.githubusercontent.com/49666604/132470876-c02337f1-ff6e-472d-b84a-f1b64acfb3e1.png)

- Run colmap point_triangulator

```bash
colmap point_triangulator \
    --database_path $PROJECT_PATH/database.db \
    --image_path  $PROJECT_PATH/images \
    --input_path  $PROJECT_PATH/inputs \
    --output_path  $PROJECT_PATH/sparse_triangulator
```
I expected that the results of 'colmap mapper' and 'colmap point_trangulator' should be same, because I used the estimated camera pose from colmap mapper. 

However, 
- **The one image's information is missing in images.bin from  'colmap point_triangulator'**
 There are only N-1 images' pose and 3D points information.
- Also the numbers of 3Dpoints(from mapper & point_trangulator) are different to each other. 
 
**2. Error with 2 images**
I followed the same procedure to new image set which contains only 2 images.
- Sparse reconstruction(using colmap mapper) was successful
- However when I run `colmap point_triangulator`, I got following error. 
![image](https://user-images.githubusercontent.com/49666604/132473559-99f5a5ae-3e1d-4f05-a4c6-567ee622ccb2.png)

Could you explain me why I got this problem?
 
I used colmap 3.7.

**Comment by mjmjeong** [comment]
I used colmap 3.7

![image](https://user-images.githubusercontent.com/49666604/132476135-e41a1153-51ad-4d26-947f-fee35340db3c.png)

**Comment by tsattler** [comment]
Regarding your points:

1. Colmap has randomized components, so the results of the point_triangulator and sparse reconstruction do not need to be the same. For the first image, did it contain many 3D points before?
2. By default, Colmap is not using feature tracks containing two images. If you want to triangulate from two images, you should set `--Mapper.tri_ignore_two_view_tracks` to 0.

**Comment by mjmjeong** [comment]
Thank you for your response.
 I solved the problem with your advice.

Also, I noticed that the last line of inputs/images.txt should be empty. That was the reason I got N-1 images after running colmap point_triangulator.  

Before
![image](https://user-images.githubusercontent.com/49666604/132633171-ad9e72f0-ecf2-45e5-ac2c-8f74d9619d70.png)

After
![image](https://user-images.githubusercontent.com/49666604/132632920-0c192674-6ec6-4da1-b862-d0bf060b53e1.png)
